### PR TITLE
remove featuregate in 1.25

### PR DESCRIPTION
**Pull request by cyclinder**
Signed-off-by: cyclinder <qifeng.guo@daocloud.io>

#### What type of PR is this?

/kind cleanup
#### What this PR does / why we need it:
remove featuregate in 1.25
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/registry/storage/csidriver/strategy.go [hunk 1]
```diff
@@ -85,8 +82,7 @@ func (csiDriverStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.
 		!utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
 		newCSIDriver.Spec.VolumeLifecycleModes = nil
 	}
-	if oldCSIDriver.Spec.FSGroupPolicy == nil &&
-		!utilfeature.DefaultFeatureGate.Enabled(features.CSIVolumeFSGroupPolicy) {
+	if oldCSIDriver.Spec.FSGroupPolicy == nil {
 		newCSIDriver.Spec.FSGroupPolicy = nil
 	}
```
Comment (pohly) [inline pkg/registry/storage/csidriver/strategy.go:87]: This looks wrong. Now that features.CSIVolumeFSGroupPolicy is always enabled, the if branch should never get taken and thus needs to be removed.

**Review by pohly — CHANGES_REQUESTED** [review]
Thanks for taking care of this.

File: pkg/registry/storage/csidriver/strategy.go (hunk 1 above)
Comment (cyclinder) [reply pkg/registry/storage/csidriver/strategy.go]: Thanks! Done.

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
/assign @lavalamp

For approval.

**Comment by fedebongio** [comment]
@lavalamp let us know if you need help assigning to someone else.
/triage accepted

**Pull request merged**